### PR TITLE
Kd login

### PR DESCRIPTION
**Pull request by cihangir**
Adds handler to kloud for logging in users

## How Has This Been Tested?

Unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by rjeczalik** [comment]
@cihangir Could you rebase please?

**Review by ppknap — APPROVED** [review]

File: go/src/koding/kites/kloud/stack/auth.go [hunk 1]
```diff
@@ -23,23 +28,57 @@ type LoginResponse struct {
...
+		k.Log.Debug("auth login called by %q with %q", r.Username, r.Args.Raw)
+
+		req, err := getLoginReq(r)
+		if err != nil {
+			return nil, err
+		}
+
+		ses, err := modelhelper.UserLogin(r.Username, req.GroupName)
+		if err != nil {
+			return nil, NewError(ErrInternalServer)
```
Comment (rjeczalik) [inline go/src/koding/kites/kloud/stack/auth.go]: Can we return sth like NewError(ErrNotAuthorized) when err == mgo.ErrNotFound?

Comment (rjeczalik) [inline go/src/koding/kites/kloud/stack/auth.go]: Can we also debug log any error that was shadowed? (the ones hidden from the user)

File: go/src/socialapi/workers/presence/client/client.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+	return &Client{
+		endpoint:    fullURL,
+		requestType: "POST",
+		reqFunc:     internalReqFunc,
+		HTTPClient: &http.Client{
+			Transport: &http.Transport{
+				// timeout only for dialing, if the remote server is not listening, stop asking earlier.
+				Dial: func(network, addr string) (net.Conn, error) {
+					return net.DialTimeout(network, addr, time.Second)
```
Comment (rjeczalik) [inline go/src/socialapi/workers/presence/client/client.go]: Alternative with contetx support (used in std):

```go
...= &http.Client{
    Transport: &http.Transport{
        DialContext: (&net.Dialer{
                Timeout: time.Second,
        }).DialContext,
    },
}
```

jfyi

File: go/src/koding/kites/kloud/kloud/kloud.go [hunk 3]
```diff
@@ -197,11 +200,16 @@ func New(conf *Config) (*Kloud, error) {
 		sess.Log.Warning(`disabling "remote.api" for stack operations`)
 	}
 
+	restClient := httputil.DefaultRestClient(conf.DebugMode)
+
+	pinger := client.NewInternal(conf.SocialProxyURL)
+	pinger.HTTPClient = restClient
```
Comment (rjeczalik) [inline go/src/koding/kites/kloud/kloud/kloud.go]: 👍

File: config/generateKonfig.coffee [hunk 4]
```diff
@@ -144,7 +144,7 @@ module.exports = (options, credentials) ->
     klientUrl: "https://s3.amazonaws.com/koding-klient/development/latest/klient.deb"
 
     credentialEndPoint: "#{socialApiProxyUrl}/credential"
-    presencePrivateEndpoint: "#{socialApiProxyUrl}/private/presence/ping"
+    socialProxyURL: "#{socialApiProxyUrl}"
```
Comment (rjeczalik) [inline config/generateKonfig.coffee]: Added with https://github.com/koding/koding/pull/9870/commits/6e6e68919ca4fc62bc3aca0bed0bab6208b00d22.

**Review by rjeczalik — APPROVED** [review]

**Pull request merged**